Admin becoming user password hint#168385051

BasilMawejje opened the pull request:
## PT Story: Admin: becoming a user  UX : needs hint about password
#### PT URL: https://www.pivotaltracker.com/story/show/168385051


## Changes proposed in this pull request:
1.  When an admin becomes a user and clicks 'Edit your profile' in the header dropdown menu, a pop up should come to the screen to remind the admin to use the admin password as the current password. This should happen when the admin sets focus to the current password field. 

2. If the admin uses the tab key to navigate to the current password field, the pop up should still be activated to remind the admin which password to use. 

## Screenshots (Optional):
<img width="1440" alt="Screenshot 2019-09-13 at 15 17 34" src="https://user-images.githubusercontent.com/26635598/64861893-d5cd6b00-d639-11e9-9735-309a8379aa0b.png">

---
## Ready for review:
@patmbolger @thesuss @weedySeaDragon

BasilMawejje committed:
Add modal to registrations edit page

BasilMawejje committed:
Add internationalization to modal text

BasilMawejje committed:
Move javascript script content to user.js in assets/javascripts

BasilMawejje committed:
Add admin_only_user_profile_edit route to path_helpers
Add admin_only_become_user route to path_helpers
Add feature file for admin becoming user password hint
Add steps for admin becoming user password hint
All tests green

weedySeaDragon commented on features/step_definitions/admin-becoming-user-password-hint_steps.rb:
```diff
@@ -0,0 +1,23 @@
+  visit admin_only_user_profile_edit_path(@user)
+end
+
+Then("I should navigate to {capture_string}") do |capture_string|
```
Hound can't parse this correctly; we don't need to do anything about this.

weedySeaDragon commented:
@patmbolger  - While I was writing cucumber tests for @BasilMawejje, it seemed to me that we have no way to tell if the current user was actually the admin that _became_ this user.

The [devise_masquerade  gem](https://github.com/oivoodoo/devise_masquerade) does all of this: allows you to 'become' ("masquerade as") a user and keeps track if the current user was an admin that 'became' this user

It has a simple method that we can use to see if the current user has been 'masqueraded':
`user_masquerade? # current user has been masqueraded (our "admin has 'become' a user")`

Instead of writing (and maintaining) our own functionality to do keep track of if an admin has 'become' a user, I think that we should use the `devise_masquerade` gem to accomplish this.  It's a well known gem that has been used for years. 

Don't know if we should create a new PT story and PR or what.  

But we should all discuss this -- here and/or during the next team meeting.

BasilMawejje committed:
Merge branch 'develop' of https://github.com/AgileVentures/shf-project into admin-becoming-user-password-hint#168385051

BasilMawejje committed:
Fix missing translations

BasilMawejje committed:
Rename files to use underscore

BasilMawejje committed:
Fix some broken semaphore tests

BasilMawejje committed:
Clean up spec background
Change admin become user path
Modify spec background to make both users admin in order to work around Pundit error
Add login_as to step file to simulate admin become action

BasilMawejje commented:
@weedySeaDragon In the spec background I made both users admins in order to work around the Pundit::NotDefinedError that comes when an admin becomes a non-admin user in the tests. All tests are green. You can review and give me feedback on the implementation.

BasilMawejje committed:
Modify spec steps to reflect admin become new user login

weedySeaDragon commented:
The system is already doing what it should

Once the admin "becomes"  a user, the system expects that the password entered is the _user's_ password.

Hence we discovered we don't need this PR and are closing it.
